utils/watchcat/files/watchcat.init

```diff
@@ -90,53 +87,36 @@ load_watchcat() {
 		logger -p user.err -t "watchcat" "reboot program $1 not started - $error"
 		return
 	}
-
+	
+	# Need to conditionally run mode functions because they have different signatures
 	case "$mode" in
-	periodic_reboot)
-		/usr/bin/watchcat.sh "periodic_reboot" "$period" "$forcedelay" &
-		logger -p user.info -t "watchcat" "started task (mode=$mode;period=$period;forcedelay=$forcedelay)"
-		;;
-	ping_reboot)
-		/usr/bin/watchcat.sh "ping_reboot" "$period" "$forcedelay" "$pinghosts" "$pingperiod" "$pingsize" &
-		logger -p user.info -t "watchcat" "started task (mode=$mode;period=$period;pinghosts=$pinghosts;pingperiod=$pingperiod;forcedelay=$forcedelay;pingsize=$pingsize)"
-		;;
-	restart_iface)
-		/usr/bin/watchcat.sh "restart_iface" "$period" "$pinghosts" "$pingperiod" "$pingsize" "$interface" "$mmifacename" &
-		logger -p user.info -t "watchcat" "started task (mode=$mode;period=$period;pinghosts=$pinghosts;pingperiod=$pingperiod;pingsize=$pingsize;interface=$interface;mmifacename=$mmifacename;unlockbands=$unlockbands)"
-		;;
-	*)
-		echo "Error starting Watchcat service. Invalid mode selection: $mode"
-		;;
+		periodic_reboot)
+			procd_open_instance "watchcat_${1}"
+			procd_set_param command /usr/bin/watchcat.sh "periodic_reboot" "$period" "$forcedelay"
+			;;
+		ping_reboot)
+			procd_open_instance "watchcat_${1}"
+			procd_set_param command /usr/bin/watchcat.sh "ping_reboot" "$period" "$forcedelay" "$pinghosts" "$pingperiod" "$pingsize"
+			;;
+		restart_iface)
+			procd_open_instance "watchcat_${1}"
+			procd_set_param command /usr/bin/watchcat.sh "restart_iface" "$period" "$pinghosts" "$pingperiod" "$pingsize" "$interface" "$mmifacename"
+			;;
+		*)
+			echo "Error starting Watchcat service. Invalid mode selection: $mode"
+			;;
 	esac
 
-	echo $! >>"${PIDFILE}.pids"
+	procd_set_param respawn ${respawn_threshold:-3600} ${respawn_timeout:-5} ${respawn_retry:-5}
+	procd_close_instance
```

Comment:
From my point of view, this is still not correct. What if we end up in the `*`? Then the functions`procd_set_param_respawn` and `procd_close_instance` are still called.